### ISSUE TITLE
BDD100K accuracy, IOU

### ISSUE THREAD

**Issue by Hydragon516** [cardwing/Codes-for-Lane-Detection#299]
First of all, thank you for sharing the good code.

In your paper, the IOU value for BDD100K is quite small as 16.02, but in my code implemented with Pytorch, the validation set IOU is measured as high as 70 or more in 1 epoch. Are there any differences in how IOU are measured? The code I used is below.

def get_miou(pred, label, num_classes=2):
    pred = pred.cpu().data.numpy().flatten()
    label = label.cpu().data.numpy().flatten()

    hist = np.bincount(num_classes * label.astype(int) + pred, minlength=num_classes ** 2)
    hist = hist.reshape(num_classes, num_classes)

    miou = np.diag(hist) / (np.sum(hist, axis=1) + np.sum(hist, axis=0) - np.diag(hist))
    miou = np.nanmean(miou)

    return miou

**Comment by cardwing** [comment]
"We decide to use pixel accuracy and IoU of lanes to evaluate the performance of different models". The IOU of lanes is measured, not including the background.

**Comment by Hydragon516** [comment]
> "We decide to use pixel accuracy and IoU of lanes to evaluate the performance of different models". The IOU of lanes is measured, not including the background.

Thank you for your kind answer. So, how is the IOU measured if the lane does not exist (i.e. the GT does not exist)? Are they excluded from the evaluation?

**Comment by cardwing** [comment]
The IoU will just be zero if GT does not exist. You can refer to train.lua for more details.
